src/App.tsx

```diff
@@ -1,26 +1,142 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import "./App.css";
+import React, { useState } from "react";
+
+
+
+interface SquareProps {
+	value: string | null;
+	onClick: () => void;
+}
+
+const Square: React.FC<SquareProps> = ({ value, onClick }) => {
+	return (
+		<button className="square" onClick={() => onClick()}>
```

Comment:
エラー出たらごめんw
ここはこうかけたりしないかな？
```suggestion
		<button className="square" onClick=onClick>
```

もしくはこう

```suggestion
		<button className="square" onClick={onClick}>
```

Reply:
こうかも？笑

```suggestion
		<button className="square" onClick={onClick()}>
```

何が伝えたいかというと、 `() => onClick()` で、元々 `onClick` の型が `() => void` と定義されてるから、 `()` はわざわざ書かなくてもいいかなと思った！

Reply:
> なぜonClickの後に()が必要ないんですか？

これの答えは

> Square()の段階ではonClickと書くだけで関数として認識されているということですか？

ほぼこれなんだけど、なんで `onClick()` ではダメで `onClick` ならいいのか考えてみるといいかも！
そして同じように `onClick(i)` と `() => onClick(i)` の違いも 👍 

調べてみて仮説持てたら、口頭でも大丈夫だから俺に聞いて〜 👌